Provider documentation fixes (#7823)
traefik/traefik: docs/content/providers/kubernetes-gateway.md

The Kubernetes Gateway API, The Experimental Way.
{: .subtitle }

Gateway API is the evolution of Kubernetes APIs that relate to `Services`, such as `Ingress`.
The Gateway API project is part of Kubernetes, working under SIG-NETWORK.

> The Kubernetes Gateway provider is a Traefik implementation of the [service APIs](https://github.com/kubernetes-sigs/service-apis)

Reviewer: ```suggestion
The Kubernetes Gateway provider is a Traefik implementation of the [Service APIs](https://kubernetes-sigs.github.io/service-apis/)
```

specifications from the Kubernetes Special Interest Groups (SIGs).

> This provider is proposed as an experimental feature and partially supports the service APIs [v0.1.0](https://github.com/kubernetes-sigs/service-apis/releases/tag/v0.1.0) specification.

Reviewer: ```suggestion
This provider is proposed as an experimental feature and partially supports the Service APIs [v0.1.0](https://github.com/kubernetes-sigs/service-apis/releases/tag/v0.1.0) specification.
```

    ```yaml tab="RBAC"
    --8<-- "content/reference/dynamic-configuration/kubernetes-gateway-rbac.yml"
    ```

> The Kubernetes Service APIs provides several [guides](https://kubernetes-sigs.github.io/service-apis/guides/) on how to use their API.

Reviewer: ```suggestion
The Kubernetes Service APIs project provides several [guides](https://kubernetes-sigs.github.io/service-apis/guides/) on how to use the APIs.
```
